src/agent/border_agent.hpp

```diff
@@ -49,6 +49,18 @@
 #include "backbone_router/backbone_agent.hpp"
 #endif
 
+#ifndef OTBR_VENDOR_NAME
+#define OTBR_VENDOR_NAME "OpenThread"
+#endif
+
+#ifndef OTBR_PRODUCT_NAME
+#define OTBR_PRODUCT_NAME "BorderRouter"
+#endif
+
+#ifndef OTBR_MESHCOP_SERVICE_NAME
+#define OTBR_MESHCOP_SERVICE_NAME OTBR_VENDOR_NAME "_" OTBR_PRODUCT_NAME
```

Comment:
There will be name conflicts if two OTBRs are placed on the same network link. mDNS Responder can auto rename the conflicted names, but avahi seems not. See https://github.com/openthread/ot-br-posix/issues/826.

Reply:
I think this issue can be regarded as independent of the current PR.

For the issue #826, I can think of two ways to fix it:
- Generating unique service names (e.g. using random digits for the suffix) before registering the service.
- Leveraging avahi's utils to handle name conflicts. There's an example in their doc: https://www.avahi.org/doxygen/html/client-publish-service_8c-example.html.